# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.cpp
# Commit: 1c45b135318d
# Full Hash: 1c45b135318d8cb519b8d58af72ab864fe6bfd62
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-11-14 17:12:31
# Regressor Bug: 1172394
# File Overlap Count: 1
# Description:
#   Bug 1172394 - Simplify MediaSink somewhat. r=padenot
#   
#   This patches does several minor things:
#   - Moves SetSink (from setSinkid) to automatic coalescing of multiple calls
#     through a Canonical/Mirror setup instead of a manual atomic counter.
# ==============================================================================

diff -r c57c41e8c39e -r 1c45b135318d dom/media/mediasink/AudioSinkWrapper.cpp
--- a/dom/media/mediasink/AudioSinkWrapper.cpp	Wed Nov 13 08:55:39 2019 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.cpp	Wed Nov 13 08:55:54 2019 +0000
@@ -21,21 +21,6 @@
   mCreator = nullptr;
 }
 
-const MediaSink::PlaybackParams& AudioSinkWrapper::GetPlaybackParams() const {
-  AssertOwnerThread();
-  return mParams;
-}
-
-void AudioSinkWrapper::SetPlaybackParams(const PlaybackParams& aParams) {
-  AssertOwnerThread();
-  if (mAudioSink) {
-    mAudioSink->SetVolume(aParams.mVolume);
-    mAudioSink->SetPlaybackRate(aParams.mPlaybackRate);
-    mAudioSink->SetPreservesPitch(aParams.mPreservesPitch);
-  }
-  mParams = aParams;
-}
-
 RefPtr<MediaSink::EndedPromise> AudioSinkWrapper::OnEnded(TrackType aType) {
   AssertOwnerThread();
   MOZ_ASSERT(mIsStarted, "Must be called after playback starts.");
@@ -154,6 +139,11 @@
   }
 }
 
+double AudioSinkWrapper::PlaybackRate() const {
+  AssertOwnerThread();
+  return mParams.mPlaybackRate;
+}
+
 nsresult AudioSinkWrapper::Start(const TimeUnit& aStartTime,
                                  const MediaInfo& aInfo) {
   AssertOwnerThread();